### PR TITLE
test: clone alert

### PR DESCRIPTION
**Pull request by neha00290**
## Summary by CodeRabbit

- **New Features**
	- Enhanced alert creation and cloning functionality with improved validation checks.
	- Added support for comprehensive search capabilities in the alert list.
	- Introduced functionality to verify user interactions with search history.

- **Bug Fixes**
	- Improved error handling and user feedback during alert cloning.
	- Enhanced validation for required fields in alert creation.

- **Tests**
	- New test cases added for alert management and search history functionalities.
	- Updated existing tests to ensure better coverage and error handling, including the addition of new scenarios for invalid inputs.

- **Chores**
	- Cleaned up the test suite by removing unused code and improving structure.

### CODE REVIEW

File: tests/ui-testing/playwright-tests/alerts.spec.js [hunk 1]
```diff
@@ -239,6 +219,162 @@ test.describe("Alerts testcases", () => {
...
+    await page.locator('[data-test="template-list-search-input"]').click();
+    await page
+        .locator('[data-test="template-list-search-input"]')
+        .fill("sanit");
+    await page.waitForTimeout(2000);
+    await page
+        .locator('[data-test="alert-template-list-sanitytemplates-delete-template"]')
+        .click();
+    await page.locator('[data-test="confirm-button"]').click();
+  });
```
[the remark this answers is not in the document]
Comment (neha00290) [reply tests/ui-testing/playwright-tests/alerts.spec.js:377]: this cannot be made short as this is a full test for creating alert which needs all above steps

File: tests/ui-testing/playwright-tests/alerts.spec.js [hunk 2]
```diff
@@ -239,6 +219,162 @@
...
+    await page.locator('[data-test="menu-link-\\/alerts-item"]').click();
+    await page.waitForTimeout(2000);
+    await page
+        .locator('[data-test="alerts-page"] div')
+        .filter({
+            hasText:
+                "AlertsDestinationsTemplateskeyboard_arrow_upkeyboard_arrow_down",
+        })
+        .first()
+        .click();
```
[the remark this answers is not in the document]
Comment (neha00290) [reply tests/ui-testing/playwright-tests/alerts.spec.js:233]: there is a filter which needs scrolling due to which this is added

File: tests/ui-testing/playwright-tests/alerts.spec.js [hunk 3]
```diff
@@ -239,6 +219,162 @@
...
+    await page.waitForResponse((response) => 
+        response.url().includes("api/default/streams?type=logs") && response.status() === 200
+    );
+
+    await page.getByLabel("Stream Name *").click();
+    await page.getByLabel("Stream Name *").fill("e2e_automate");
+    
+    await page.getByText("e2e_automate").click();
+    await page.waitForTimeout(2000);
+    await page
```
[the remark this answers is not in the document]
Comment (neha00290) [reply tests/ui-testing/playwright-tests/alerts.spec.js:312]: will fix it as need to rework on this test

File: tests/ui-testing/playwright-tests/sanity.spec.js [hunk 4]
```diff
@@ -495,6 +495,31 @@ test.describe("Sanity testcases", () => {
       )
       .click();
     await page.locator('[data-test="add-alert-submit-btn"]').click();
+
+    // Clone the alert
+    await page.locator('[data-test="alert-clone"]').click(); // Ensure this selector is correct
+    await page.getByLabel('Alert Name').click();
+    await page.getByLabel('Alert Name').fill('test-clone');
+    await page.locator('[data-test="to-be-clone-alert-name"]').click()
+    await page.locator('[data-test="to-be-clone-alert-name"]').fill('test-clone');
```
[the remark this answers is not in the document]
Comment (neha00290) [reply tests/ui-testing/playwright-tests/sanity.spec.js:505]: same locator is used for both clone alert name as well

File: tests/ui-testing/playwright-tests/sanity.spec.js [hunk 5]
```diff
@@ -491,17 +492,42 @@
...
+    await page.locator('[data-test="to-be-clone-stream-type"]').click();
+    
+    await page.getByRole('option', { name: 'logs' }).locator('div').nth(2).click();
+    await page.locator('[data-test="to-be-clone-stream-name"]').click();
+    await page.locator('[data-test="to-be-clone-stream-name"]').fill('e2e_automate');
+    await page.waitForTimeout(2000);
+    await page.getByRole('option', { name: 'e2e_automate' }).click({force:true});
```
[the remark this answers is not in the document]
Comment (neha00290) [reply tests/ui-testing/playwright-tests/sanity.spec.js:512]: need to check the ingested data which is with e2e_automate

File: tests/ui-testing/playwright-tests/sanity.spec.js [hunk 6]
```diff
@@ -491,17 +492,42 @@ test.describe("Sanity testcases", () => {
...
+    await page.waitForTimeout(2000);
+    await page.locator('[data-test="clone-alert-submit-btn"]').click();
+
```
[the remark this answers is not in the document]
Comment (neha00290) [reply tests/ui-testing/playwright-tests/sanity.spec.js]: added assertion

File: tests/ui-testing/playwright-tests/alerts.spec.js [hunk 7]
```diff
@@ -239,6 +219,162 @@
...
+    await page.locator('[data-test="to-be-clone-stream-name"').click();
+    await page.locator('[data-test="to-be-clone-stream-name"').fill('e2e_automate');
+    await page.getByRole('option', { name: 'e2e_automate' }).click();
+    await page.locator('[data-test="clone-alert-submit-btn"]').click();
+    // await page.locator('.q-form > label:nth-child(2) > .q-field__inner > .q-field__control > .q-field__control-container > .q-field__native').click();
+    // await page.getByRole('option', { name: 'logs' }).locator('div').nth(2).click();
+    // await page.getByLabel('Stream Name').click();
+    // await page.getByLabel('Stream Name').fill('e2e_automate');
+    // await page.getByRole('option', { name: 'e2e_automate' }).click();
+    // await page.locator('[data-test="add-alert-submit-btn"]').click();
```
[the remark this answers is not in the document]
Comment (neha00290) [reply tests/ui-testing/playwright-tests/alerts.spec.js]: removed

**Review by kethesainikhil — APPROVED** [review]

**Review by nikhilsaikethe — APPROVED** [review]

**Pull request merged**